spec/requests/generic_object_definitions_spec.rb

```diff
@@ -0,0 +1,313 @@
+# rubocop:disable Style/WordArray
+RSpec.describe 'GenericObjectDefinitions API' do
+  let(:object_def) { FactoryGirl.create(:generic_object_definition, :name => 'foo') }
+
+  describe 'GET /api/generic_object_definitions' do
+    it 'does not list object definitions without an appropriate role' do
+      api_basic_authorize
+
+      run_get(generic_object_definitions_url)
+
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it 'lists all generic object definitions with an appropriate role' do
+      api_basic_authorize collection_action_identifier(:generic_object_definitions, :read, :get)
+
+      expected = {
+        'count'     => 1,
+        'subcount'  => 1,
+        'name'      => 'generic_object_definitions',
+        'resources' => [
+          hash_including('href' => a_string_matching(generic_object_definitions_url(object_def.compressed_id)))
+        ]
+      }
+      run_get(generic_object_definitions_url)
```

Comment:
Also, tiny nit, but can you group all the verification code together and separately from the execution step(s)? 🍨